### PR TITLE
feat: Select payment methods as default in the payment settings

### PR DESCRIPTION
**Pull request by SGrueber**
## PR Type

[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: 

## What Is the Current Behavior?
Only payment instruments like credit cards can be set as preferred payment methods in the myAccount payment settings.

Issue Number: Closes #

## What Is the New Behavior?
Also payment methods that don't require parameters like invoice can be set as default payment method.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

## Other Information

[AB#76608](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/76608)

### CODE REVIEW

**Review by iwiederhold — CHANGES_REQUESTED** [review]
- Strange behavior when loading. Please check this process.
- Different panel heights for single line texts at the Preferred payment panel.
- Could we use the same spaces inside the checkout payment panels?

**Review by iwiederhold — APPROVED** [review]
Nice! And good cooperation!

**Review by M-Behr — CHANGES_REQUESTED** [review]
It works very well 👍 Yesterday we decided together to make small changes:

- [x] Show another success message, when selecting "No preferred payment method". Something like: "You have no longer a preferred payment method selected."
- [x] Replace trash icons with "Delete" link for a consistent UI between Checkout and My Account

**Review by M-Behr — DISMISSED** [review]
Perfect, thank you!

File: src/assets/i18n/de_DE.json [hunk 1]
```diff
@@ -275,17 +275,19 @@
...
   "account.password.label": "Aktuelles Kennwort",
-  "account.payment.further_payments.heading": "Weitere gespeicherte Zahlungsmittel",
+  "account.payment.further_payments.heading": "Weitere Zahlungsmethoden",
   "account.payment.heading": "Gespeicherte Zahlungsdaten",
   "account.payment.link": "Zahlung",
-  "account.payment.message": "Wählen Sie ein bevorzugtes Zahlungsmittel für eine schnellere Zahlung bei zukünftigen Einkäufen.",
-  "account.payment.no_entries": "Derzeit haben Sie keine gespeicherten Zahlungsmittel.",
-  "account.payment.payment_created.message": "Ihr bevorzugtes Zahlungsmittel wurde in Ihrem Konto gespeichert.",
+  "account.payment.message": "Wählen Sie eine bevorzugte Zahlungsmethode für eine schnellere Zahlung bei zukünftigen Einkäufen.",
+  "account.payment.no_entries": "Derzeit haben Sie keine Zahlungsmethoden.",
```
Comment (mglatter) [inline src/assets/i18n/de_DE.json:282]: Fehlt hier nicht noch "gespeicherten"?

File: src/assets/i18n/en_US.json [hunk 2]
```diff
@@ -275,17 +275,19 @@
...
   "account.password.label": "Current Password",
-  "account.payment.further_payments.heading": "Further Saved Payment Instruments",
+  "account.payment.further_payments.heading": "Further Payment Methods",
   "account.payment.heading": "Saved Payment Information",
   "account.payment.link": "Payment",
-  "account.payment.message": "Select a preferred payment instrument for quicker checkout on future purchases.",
-  "account.payment.no_entries": "Currently, you do not have any saved payment instruments.",
-  "account.payment.payment_created.message": "Your preferred payment instrument has been saved to your account.",
+  "account.payment.message": "Select a preferred payment method for quicker checkout on future purchases.",
+  "account.payment.no_entries": "Currently, you do not have any payment methods.",
```
Comment (mglatter) [inline src/assets/i18n/en_US.json:282]: No comma. See comment for DE: Is "saved" missing?
Else maybe write "Currently there are no payment methods."

File: src/assets/i18n/en_US.json [hunk 3]
```diff
@@ -275,17 +275,19 @@
...
+  "account.payment.no_preferred.message": "You have no longer a preferred payment method selected.",
```
Comment (mglatter) [inline src/assets/i18n/en_US.json:283]: "You no longer have a preferred payment method selected."

File: src/assets/i18n/en_US.json [hunk 4]
```diff
@@ -275,17 +275,19 @@
...
+  "account.payment.no_preferred_method": "No preferred payment method",
+  "account.payment.payment_created.message": "Your preferred payment method has been saved to your account.",
   "account.payment.payment_deleted.message": "Your payment instrument has been deleted.",
```
Comment (mglatter) [inline src/assets/i18n/en_US.json:286]: See DE comment - Does "payment instrument" need to be changed as well?

File: src/assets/i18n/fr_FR.json [hunk 5]
```diff
@@ -275,17 +275,19 @@
...
+  "account.payment.no_preferred.message": "Vous n'avez plus sélectionné de mode de paiement préféré.",
+  "account.payment.no_preferred_method": "Pas de mode de paiement par défaut",
+  "account.payment.payment_created.message": "Votre mode de paiement par défaut a été enregistré sur votre compte.",
   "account.payment.payment_deleted.message": "Votre moyen de paiement a été supprimé.",
-  "account.payment.preferred.link": "Enregistrer comme moyen de paiement par défaut",
-  "account.payment.preferred_method": "Moyen de paiement par défaut",
+  "account.payment.preferred_method": "Votre mode de paiement par défaut",
   "account.payment.sepa_accepted_on": "Accepté le",
   "account.payment.sepa_mandate_Reference": "Référence du mandat",
+  "account.payment.standard_methods": "Méthodes de paiement standard",
```
Comment (mglatter) [inline src/assets/i18n/fr_FR.json:290]: Please have Nicolas review the whole FR part. I have the feeling that this must be "Méthodes de paiement standards" but he will be sure about the correct grammar. ;-)

**Review by mglatter — CHANGES_REQUESTED** [review]
Please ask Nicolas for review of fr_FR.json

File: src/assets/i18n/fr_FR.json (hunk 5 above)
Comment (NicolasGrizard) [reply src/assets/i18n/fr_FR.json:290]: "modes de paiement" is the correct wording

Comment (NicolasGrizard) [inline src/assets/i18n/fr_FR.json:290]: should be "Modes de paiement standard"

**Review by NicolasGrizard — CHANGES_REQUESTED** [review]
line 290 of fr_FR.json should be "Modes de paiement standard"

**Review by MaxKless — APPROVED** [review]

**Pull request merged**